client/src/services/informationManager.ts

```diff
@@ -0,0 +1,155 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+import { MsAlertModal, MsAlertModalConfig, MsReportTheme } from '@/components/core';
+import { ToastManager } from '@/services/toastManager';
+import { modalController } from '@ionic/vue';
+import { v4 as uuid4 } from 'uuid';
+import { NotificationManager } from '@/services/notificationManager';
+
+export const InformationKey = 'information';
+
+export enum InformationLevel {
+  Info = 'INFO',
+  Success = 'SUCCESS',
+  Warning = 'WARNING',
+  Error = 'ERROR',
+}
+
+export interface IInformation {
```

Comment:
What's the point of this interface?

Reply:
Previously I was exporting it instead of the class, not anymore, however it's more readable to have the interface instead of the inline interface declaration in the already consistent class declaration.

Reply:
I meant that the goal of an interface is to abstract something we can have multiple of.
```typescript
interface Vehicle { wheels: number; }

class Car extends Vehicle { this.wheels = 4; }
class Bicycle extends Vehicle { this.wheels = 2; }

function doSomething(v: Vehicle) {}
```
Here it seems that only the `Information` class uses it, it's never used anywhere else.

Reply:
interface is not only meant to abstract things, in TS its first purpose is to type data-structured types, here we were already using an interface without making it explicit:
```
export class Information {
  id: string;
  message: string;
  level: InformationLevel;
  data?: object;
  constructor({ message, level, data = {} }: {
  message: string;
  level: InformationLevel;
  data?: object
}) {
    this.id = uuid4();
    this.message = message;
    this.level = level;
    this.data = data;
  }
}
```
the part after the `:` in the constructor is an interface even without declaring it, I just found more readable and simple to maintain with the interface declared separetely.

Reply:
But it adds complexity needlessly. `IInformation` is not used anywhere else, and just defines a few attributes that are redeclared anyway on `Information`.

In your example, I don't see the difference with
```typescript
export class Information {
  id: string;
  message: string;
  level: InformationLevel;
  data?: object;

  constructor(message: string, level: InformationLevel, data = {}) {
    this.id = uuid4();
    this.message = message;
    this.level = level;
    this.data = data;
  }
}
```
but this seems easier to read.

Reply:
We agreed on this format, it's easier to use and offer the possibility to create Notifications without caring about parameters order.